src/Keboola/DbExtractor/Configuration/ConfigRowDefinition.php

```diff
@@ -79,9 +64,74 @@ public function getConfigTreeBuilder(): TreeBuilder
             ->end();
         // @formatter:on
 
+        // Defined Query and Table node at the same time
+        $rootNode->validate()
+            ->ifTrue(function ($v) {
+                return (isset($v['query']) && isset($v['table']));
+            })
+            ->thenInvalid('Both table and query cannot be set together.')
+            ->end();
+
+        // Undefined Query and Table node
+        $rootNode->validate()
+            ->ifTrue(function ($v) {
+                return (!isset($v['query']) && !isset($v['table']));
+            })
+            ->thenInvalid('One of table or query is required.')
+            ->end();
+
+        // Defined Table node contains required fields
+        $rootNode->validate()
```

Comment:
nejde tohle validovat normálně nepovinným nodem? Ten už může mít povinné childy a jeho "povinnost" už máš otestovanou v  `        // Undefined Query and Table node`

Reply:
Vidis, to by melo jit.